forte/processors/data_augment/text_generation_augment_processor.py

```diff
@@ -0,0 +1,169 @@
+        Each pair of source and target document is linked with a MultiPackLink.
+        """
+        if not self.augmenter:
+            raise KeyError("The processor has not been assigned an augmenter!")
+
+        input_pack: DataPack = multipack.get_pack(self.configs.input_pack_name)
+        output_pack: DataPack = multipack.get_pack(self.configs.output_pack_name)
+        self._create_multipack_link(multipack, input_pack, output_pack)
+
+        # Create aug_num pairs of augmented source & target datapacks
+        for i in range(self.configs.aug_num):
+            aug_input_pack: DataPack = self._process_pack(multipack.get_pack(self.configs.input_pack_name))
+            aug_output_pack: DataPack = self._process_pack(multipack.get_pack(self.configs.output_pack_name))
+            multipack.update_pack({
+                "{}_{}".format(self.configs.aug_input_pack_name, str(i)): aug_input_pack,
+                "{}_{}".format(self.configs.aug_output_pack_name, str(i)): aug_output_pack,
+            })
+            # Create a link between the source and target.
+            self._create_multipack_link(multipack, aug_input_pack, aug_output_pack)
+
+    def _create_multipack_link(self, multipack: MultiPack, src: DataPack, tgt: DataPack):
+        r"""
+        Create a multipack link between the source and target document
+        within a multipack.
+        """
+        src_docs: List[Document] = list(src.get(Document))
+        tgt_docs: List[Document] = list(tgt.get(Document))
+        # If the document is not included in the datapack, create a new one.
+        if len(src_docs) == 0:
+            src_docs.append(Document(src, 0, len(src.text)))
+        if len(tgt_docs) == 0:
+            tgt_docs.append(Document(tgt, 0, len(tgt.text)))
+
+        # The datpack should only contains one document.
+        assert len(src_docs) == 1
+        assert len(tgt_docs) == 1
+        src_doc: Document = src_docs[0]
+        tgt_doc: Document = tgt_docs[0]
+        # Link the source and target document.
+        cross_link = MultiPackLink(multipack, src_doc, tgt_doc)
+        multipack.add_entry(cross_link)
+
+
+    def _process_pack(self, pack: DataPack) -> DataPack:
+        r"""
+        This function process a single datapack with an augmenter.
+        It processes one sentence at a time.
+        Args:
+            pack: a datapack with original texts and annotations
+        Returns:
+            a datapack with augmented texts and annotations
+        """
+        data_pack: DataPack = DataPack()
+        if len(pack.text) == 0:
+            return data_pack
+        pack_text: str = ""
+
+        replacement_level: str = self.configs.replacement_level
+        replacement_prob: float = self.configs.replacement_prob
+
+        for sent in pack.get(Sentence):
+            sent_text: str = sent.text
+            # Replace the whole sentence.
+            if replacement_level == 'sentence':
```

Comment:
If you use the entry class name as level then you don't have to create all if elses.